lib/open_api/serializer.rb

```diff
@@ -0,0 +1,14 @@
+module OpenApi
+  module Serializer
+    def as_json(arg = {})
+      return super unless kind_of?(ApplicationRecord)
+
+      super.tap do |hash|
+        hash.each_key do |attr_key|
+          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
+          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
+        end
+      end.compact!
```

Comment:
`compact` will remove those unset `nil` attributes, are we sure they are not needed anymore, like UI side?

Reply:
I was going to ask for feedback on this.  I was following topology here but my initial thought is to remove the `compact!` call and allow those values to be returned.

@syncrou @mkanoor @bzwei Any thoughts?

Reply:
I am leaning to remove `nils`. We can revisit this if we see the case in the future where nil value is needed.

Reply:
oops. I saw you chose the opposite. I guess it is ok too.